astropy/timeseries/tests/test_downsample.py

```diff
@@ -35,35 +36,96 @@ def test_timeseries_invalid():
         aggregate_downsample(None)
     assert exc.value.args[0] == ("time_series should be a TimeSeries")
 
+
+def test_time_bin_invalid():
+
+    # Make sure things crash when time_bin_* is passed incorrectly.
+
+    with pytest.raises(TypeError) as exc:
+        aggregate_downsample(TimeSeries(), time_bin_size=1)
+    assert exc.value.args[0] == ("'time_bin_size' should be a Quantity or a TimeDelta")
```

Comment:
```suggestion
    # Make sure to raise the right exception when time_bin_* is passed incorrectly.

    with pytest.raises(TypeError, match=r"'time_bin_size' should be a Quantity or a TimeDelta"):
        aggregate_downsample(TimeSeries(), time_bin_size=1)
```
I think "things crash" may be more commonly understood as the interpreter segfaulting etc. ;-)

This and other exceptions could be more concisely checked using the `match` option, but I don't think we have a specific policy on that, so it remains somewhat up to taste.